python/hsfs/util.py

```diff
@@ -143,12 +142,21 @@ def get_hudi_datestr_from_timestamp(timestamp):
 def convert_event_time_to_timestamp(event_time, time_zone=timezone.utc):
     if event_time is None:
         return None
-    if isinstance(event_time, str):
+    elif isinstance(event_time, str):
         if event_time:
             return get_timestamp_from_date_string(event_time, time_zone)
-    if isinstance(event_time, pd._libs.tslibs.timestamps.Timestamp):
-        # pandas.timestamp represents millisecond in decimal
+    elif isinstance(event_time, pd._libs.tslibs.timestamps.Timestamp):
+        # convert to unix epoch time in milliseconds.
         return event_time.timestamp() * 1000
+    elif isinstance(event_time, datetime):
+        # convert to unix epoch time in milliseconds.
+        return int(event_time.replace(tzinfo=time_zone).timestamp() * 1000)
```

Comment:
In the timezone PR here https://github.com/logicalclocks/feature-store-api/pull/789, I only set the timezone to UTC when `datetime.tzinfo` is None, otherwise I keep the timezone information. Let's do the same here.

Reply:
done